README.md

```diff
@@ -0,0 +1,72 @@
+
+# Lua.js
+
+[ALPHA] A non-crappy Lua 5.1 - 5.4 parsing and compiling library.
+
+## Description
+
+Lua.js is a modular and fast Lua interpreter written in TypeScript and JavaScript. It supports parsing and compiling Lua versions 5.1 to 5.4.
+
+## Features
+
+- Supports Lua 5.1 - 5.4
+- Modular design
+- Written in TypeScript for type safety
+- Fast and efficient parsing and compiling
+
+## Installation
+
+To install Lua.js, use npm:
+
+```bash
+npm install lua-js
+```
+
+## Usage
+
+Here's a basic example of how to use Lua.js:
+
+```typescript
+import { Lexer, Parser, Compiler } from 'lua-js';
+
+const luaCode = `
+print("Hello, Lua!")
+`;
+
+const lexer = new Lexer(luaCode);
+const parser = new Parser(lexer);
```

Comment:
You're passing Lexer object to parser, instead you should pass tokens generated from the Lexer's `lex` method -- `lexer.lex()`

Reply:
i dont know what you want bro :sob:

Reply:
> i dont know what you want bro 😭

READ. IT. AGAIN.

HOW CAN YOU POSSIBLY NOT UNDERSTAND THIS?! LEXER IS A COMPONENT THAT TAKES YOUR CODE AND BREAKS IT DOWN INTO TOKENS. IT'S NOT JUST SOME RANDOM OBJECT YOU CAN PASS AROUND LIKE A TOY! IT CONVERTS CODE INTO ABSTRACT SYNTAX TREES, FOR GOODNESS' SAKE!

WHY IN THE WORLD ARE YOU PASSING AN INSTANCE OF **LEXER** TO **PARSER**?! DO YOU EVEN REALIZE HOW ABSURD THAT IS?! IT'S LIKE TRYING TO USE A HAMMER TO FIX A COMPUTER! WHAT DO YOU THINK IS GOING TO HAPPEN? DO YOU THINK THE PARSER IS JUST GOING TO FIGURE IT OUT ON ITS OWN? IT DOESN'T WORK THAT WAY! GET IT TOGETHER! THIS ISN'T ROCKET SCIENCE! IT'S FUNDAMENTAL! IF YOU CAN'T EVEN GET THIS RIGHT, HOW DO YOU EXPECT TO HANDLE ANYTHING MORE COMPLEX? THIS IS FRUSTRATING BEYOND BELIEF! GET A GRIP AND FIX YOUR CODE!

ARE YOU MAD?!

Reply:
```ts

const lexer = new Lexer(luaCode);
const tokens = lexer.lex();
const parser = new Parser(tokens);
```

Reply:
bet